cli/command/container/attach.go

```diff
@@ -70,8 +70,11 @@ func NewAttachCommand(dockerCli command.Cli) *cobra.Command {
 	return cmd
 }
 
-func runAttach(dockerCli command.Cli, opts *attachOptions) error {
-	ctx := context.Background()
+func RunAttach(dockerCli command.Cli, opts *attachOptions) error {
+	return RunAttachWithContext(context.Background(), dockerCli, opts)
+}
+
+func RunAttachWithContext(ctx context.Context, dockerCli command.Cli, opts *attachOptions) error {
```

Comment:
Perhaps we only need one; we can move the `context.Background()` to `NewAttachCommand`, and make the (now exported) `RunAttach` accept a context.

Reply:
I'm also looking if we can change this function to accept just the `apiClient`, but looks like that may need more work.

Reply:
Oh! Actually, I don't think just exporting this will help you, because `attachOptions` is not exported, so we may need a function that takes `container.AttachOptions` instead (but that means we need some way to set all the required options in it, which looks to be somewhat involved still.

Reply:
I tried to limit changes to the strict minimum :) 
I wonder why we use a `context.Background()` here and not the cobra `cmd.Context()`

Reply:
> change this function to accept just the apiClient

this indeed would have much more impacts, as `dockerCli` is passed to a bunch of public functions, whenever only APIClient or Streams is actually required. This would require a more significant deprecation/newFunc dance :P

Reply:
> because attachOptions is not exported

I've been to fast here indeed :)
also exported AttachOptions (same way I did for `ExecOptions` on https://github.com/docker/cli/commit/2d268392d1354a3cca68dac91a0cf93638a55d0d)

Reply:
> I wonder why we use a context.Background() here and not the cobra cmd.Context()

My best guess; because it originally wasn't there when this code was written (it was added in 2020, and likely code wasn't updated at the time)